token.go

```diff
@@ -82,9 +84,12 @@ func (t *Token) SigningString() (string, error) {
 	return strings.Join(parts, "."), nil
 }
 
-// Parse parses, validates, and returns a token.
-// keyFunc will receive the parsed token and should return the key for validating.
-// If everything is kosher, err will be nil
+// Parse parses, validates, verifies the signature and returns the parsed token.
+// keyFunc will receive the parsed token and should return the cryptographic key
+// for verifying the signature.
+// keyFunc should validate the 'alg' claim in the token matches the expected algorithm.
```

Comment:
Uhm to be honest. I think this is not the correct place for this documentation or the correct way to use `keyFunc`.

The general idea is to supply a slice of `ValidMethods` to the `Parser` object, so that the matching algorithms can be checked before key func is ever reached. The idea of `keyFunc` is purely to provide a key. Sure it can be used for further custom validation, but that is not the primary point of it.

https://github.com/golang-jwt/jwt/blob/2ebb50f957d606de5909fcf9ed49f9af3bc35e97/parser.go#L30

Reply:
That's a good point. IMO, the `Parse` functions defined here https://github.com/golang-jwt/jwt/blob/main/token.go#L88-L94 are partly the problem. They compel the application developer to create a token with a default parser that has a nil `ValidMethods`.
I was wondering if these functions should be marked deprecated. I.e. encourage the developer to create a parser with `ValidMethods`.

Reply:
I am working on a branch that features a similar syntax as the original `v4` branch, introducing parser options for an easier way to configure a parser. I just have to see whether we can do that completely backwards compatible or if this will be the beginning of the v5 branch.